common/app/components/Player.tsx

```diff
@@ -720,6 +789,7 @@ const Player = React.memo(function Player({
                 clock.stop();
             }
 
+            pendingTimeRef.current = 0;
```

Comment:
Can you make the name of this variable more clear? Maybe `pendingAutoRepeatTargetTimestamp`?
Also, is it possible to centralize the resetting of this timestamp? It looks like you're resetting it any time the user intentionally seeks somewhere, which is easier for devs to figure out when they make changes here, without understanding the purpose of this variable.